src/test/java/org/fogbowcloud/manager/core/threads/TestAttendSpawningOrdersThread.java

```diff
@@ -0,0 +1,224 @@
+import org.fogbowcloud.manager.core.models.orders.instances.ComputeOrderInstance;
+import org.fogbowcloud.manager.core.models.orders.instances.InstanceState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.fogbowcloud.manager.core.plugins.ComputePlugin;
+import org.fogbowcloud.manager.core.utils.SshConnectivityUtil;
+import org.fogbowcloud.manager.core.utils.TunnelingServiceUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class TestAttendSpawningOrdersThread {
+
+	private static final String DEFAULT_ORDER_INSTANCE_ID = "1";
+	
+	private AttendSpawningOrdersThread attendSpawningOrdersThread;
+	private ComputePlugin computePlugin;
+	private Long sleepTime;
+	private Properties properties;
+	
+	@Before
+	public void setUp() {		
+		this.computePlugin = Mockito.mock(ComputePlugin.class);		
+		this.attendSpawningOrdersThread = Mockito.spy(new AttendSpawningOrdersThread(this.computePlugin, this.sleepTime));		
+		this.properties = new Properties();
+		this.properties.put(ConfigurationConstants.XMPP_ID_KEY, ".");
+	}
+	
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testProcesseComputeOrderInstanceActive() {
+		Order order = createOrder();
+		SynchronizedDoublyLinkedList list = addInSpawningOrderList(order);
+		order = list.getNext();
+		
+		OrderInstance orderInstance = new OrderInstance(DEFAULT_ORDER_INSTANCE_ID);
+		order.setOrderInstance(orderInstance);
+		
+		ComputeOrderInstance computeOrderInstance = Mockito.spy(ComputeOrderInstance.class);
+		computeOrderInstance.setState(InstanceState.ACTIVE);
+		Mockito.when(this.computePlugin.getInstance(Mockito.any(Token.class), Mockito.eq(DEFAULT_ORDER_INSTANCE_ID))).thenReturn(computeOrderInstance);
+		
+		TunnelingServiceUtil tunnelingService = Mockito.mock(TunnelingServiceUtil.class);
+		this.attendSpawningOrdersThread.setTunnelingService(tunnelingService);
+		Mockito.doNothing().when(computeOrderInstance).setExternalServiceAddresses(Mockito.anyMap());
+		
+		SshConnectivityUtil sshConnectivity = Mockito.mock(SshConnectivityUtil.class);
+		this.attendSpawningOrdersThread.setSshConnectivity(sshConnectivity);
+		Mockito.when(sshConnectivity.checkSSHConnectivity(computeOrderInstance)).thenReturn(true);
+		
+		Assert.assertNull(SharedOrderHolders.getInstance().getFulfilledOrdersList().getNext());
+		this.attendSpawningOrdersThread.processSpawningOrder(order);
+		Assert.assertNull(SharedOrderHolders.getInstance().getSpawningOrdersList().getNext());
+		
+		Order test = SharedOrderHolders.getInstance().getFulfilledOrdersList().getNext();
+		assertNotNull(test);
+		Assert.assertEquals(order.getOrderInstance(), test.getOrderInstance());
+		Assert.assertEquals(OrderState.FULFILLED, test.getOrderState());
+	}
+	
+	@Test
+	public void testGetExternalServiceAddressesConfiguredToThrowException() {
```

Comment:
I do not understand this test main goal, by the fact that after calling the method processSpawningOrder there is no end checks. In this test case the Order will continue in the Spawning state right?